cmd/downloads/config/downloads_config.go

```diff
@@ -0,0 +1,355 @@
+                <li><a href="{{ .URL }}">license</a></li>
+            {{ end }}
+        {{ end }}
+	</ul>
+	{{ end }}
+</body>
+</html>`
+
+// load artifacts to be served from given path
+func loadArtifacts(path string) ([]ArtifactSpec, error) {
+	specs := ArtifactsRoot{}
+	// open the JSON spec config file
+	fileBytes, err := os.ReadFile(path)
+	if err != nil {
+		return nil, err
+	}
+
+	// unmarshal the YAML file into the ArtifactsRoot object
+	err = yaml.Unmarshal(fileBytes, &specs)
+	if err != nil {
+		return nil, err
+	}
+
+	// check if the specs are empty
+	if len(specs.Artifacts) == 0 {
+		return nil, fmt.Errorf("no artifacts to serve")
+	}
+
+	return specs.Artifacts, nil
+}
+
+// NewArtifactsConfig creates a new ArtifactsConfig object
+func NewArtifactsConfig(port int, specsFilePath string) (*ArtifactsConfig, error) {
+	tempDir, err := os.MkdirTemp("", "artifacts")
+	if err != nil {
+		return nil, err
+	}
+
+	specs, err := loadArtifacts(specsFilePath)
+	if err != nil {
+		return nil, err
+	}
+
+	templateHTML, err := template.New("artifacts").Parse(templateStringHTML)
+	if err != nil {
+		return nil, err
+	}
+
+	artifactsConfig := ArtifactsConfig{
+		Port:         fmt.Sprintf("%d", port),
+		Spec:         specs,
+		TempDir:      tempDir,
+		TemplateHTML: templateHTML,
+	}
+	err = artifactsConfig.setupArtifactsDirectory()
+	if err != nil {
+		return nil, err
+	}
+	artifactsConfig.PathToOCLicense = "/usr/share/openshift/LICENSE"
+	klog.Infof("Artifacts server configuration initialized, serving from %s", tempDir)
+
+	return &artifactsConfig, nil
+}
+
+func addFileToTar(tw *tar.Writer, filename string) error {
+	// Open the archive
+	file, err := os.Open(filename)
+	if err != nil {
+		return err
+	}
+	defer file.Close()
+
+	// Get file information
+	info, err := file.Stat()
+	if err != nil {
+		return err
+	}
+
+	// Create a tar Header from the FileInfo data
+	header, err := tar.FileInfoHeader(info, info.Name())
+	if err != nil {
+		return err
+	}
+
+	// Write file header to the tar archive
+	err = tw.WriteHeader(header)
+	if err != nil {
+		return err
+	}
+
+	// Copy file content to tar archive
+	_, err = io.Copy(tw, file)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func addFileToZip(zw *zip.Writer, filename string) error {
+	// Open the archive
+	file, err := os.Open(filename)
+	if err != nil {
+		return err
+	}
+	defer file.Close()
+
+	// Get file information
+	info, err := file.Stat()
+	if err != nil {
+		return err
+	}
+
+	// Create a zip header based on the file's information
+	header, err := zip.FileInfoHeader(info)
+	if err != nil {
+		return err
+	}
+
+	// Create a new writer for the file in the zip archive
+	writer, err := zw.CreateHeader(header)
+	if err != nil {
+		return err
+	}
+
+	// Copy the file content to the zip writer
+	_, err = io.Copy(writer, file)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// create archives for the target file
+func createArchives(pathToTargetFile string) error {
```

Comment:
nit: ideally if this function was called twice, once for tar, once for zip. But thats just small detail.